docs/common-sdk-requirements.md

```diff
@@ -0,0 +1,43 @@
+# Requirements for the Common SDK infrastructure
+The purpose of this doc is to have a place to list and discuss all requirements to the new modular SDK infrastructure (also informally called "the Common SDK infrastructure") before creating an official design.
+
+The information listed here was primarily sourced in a few meetings/discussions/whiteboardings we had before.
+
+## Terminology
+* _Common SDK_ is the new modular SDK. 
+* _Service_ is a single building unit (module) of the new modular SDK. TODO: please let me know if another term is desired for this.
+
+## Requirements
+### Requirements to the SDK
```

Comment:
We should support programatic generation and download of the SDK -- something like : 
GenerateSDK.exe <configJson> 

The Config json could be sropped with the SDK such that if someone includes it in their build chain/ teamcity -- we can regenerate the SDK with same configurations for them.

Reply:
This is an interesting idea! We should think about this. While SDK generation is already a scriptable command-line friendly process, making it a product that can be downloaded and used by client as a script or library so that they can use it, for example, in their automation/build systems sounds powerful.